app/src/main/java/ch/epfl/culturequest/ui/scan/ScanFragment.java

```diff
@@ -160,6 +162,7 @@ public void onSurfaceTextureUpdated(@NonNull SurfaceTexture surfaceTexture) {
                     .setNegativeButton("Wait", ((dialog, which) -> dialog.cancel()))
                     .setPositiveButton("Cancel scan", (dialog, id) -> {
                         loadingAnimation.stopLoading();
+                        FireStorage.deleteImage(scannedImageUrl);
```

Comment:
If I'm not mistaken, at this stage, we can't be sure that the whole scanned image processing has been performed, so `scannedImageUrl` could still be null. You could simply solve that by performing a null check in `deleteImage` which would make the function return instantly, or implement the check right before `deleteImage` is being called.